exporter/metric/metric.go

```diff
@@ -80,15 +79,8 @@ func newMetricExporter(o *options) (*metricExporter, error) {
 		return nil, errBlankProjectID
 	}
 
-	clientOpts := append(o.MonitoringClientOptions, option.WithUserAgent(userAgent))
-	ctx := o.Context
-	if ctx == nil {
-		ctx = context.Background()
-	}
-	client, err := monitoring.NewMetricClient(ctx, clientOpts...)
-	if err != nil {
-		return nil, err
-	}
+	cloudMock := cloudmock.NewCloudMock()
```

Comment:
This is a production code, not test. You shouldn't be using mocks in the production code, so please revert this change.

Instead you should probably have a test invoking `newMetricExporter` with `clientOpts` similar to approach in #66. @the-ericwang35 could you please assist Mohamed with that if you have time?

Reply:
> This is a production code, not test. You shouldn't be using mocks in the production code, so please revert this change.
> 
> Instead you should probably have a test invoking `newMetricExporter` with `clientOpts` similar to approach in #66.

I agree, as stated in the conversation [here](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/pull/67#discussion_r456017469). I'm assuming you missed the change/my comment when you suggested merging the code as is. I reverted all the changes on `metric.go`, and instead used `newMetricExporter` where appropriate.